pkgs/applications/networking/instant-messengers/discord/default.nix

```diff
@@ -62,10 +62,25 @@ let
   };
   package = if stdenv.isLinux then ./linux.nix else ./darwin.nix;
 
-  openasar = if withOpenASAR then callPackage ./openasar.nix { } else null;
+  # OpenAsar fails with default unzip, throwing  "lchmod (file attributes) error: Operation not supported"
+  unzipFix =
+    if stdenv.isLinux then
+      unzip.overrideAttrs (oldAttrs: {
+        buildFlags = oldAttrs.buildFlags ++ [ "LOCAL_UNZIP=-DNO_LCHMOD" ];
+      })
+    else
+      unzip;
+
+  openasar = callPackage ./openasar.nix { unzip = unzipFix; };
 
   packages = (builtins.mapAttrs
-    (_: value: callPackage package (value // { inherit src version openasar; meta = meta // { mainProgram = value.binaryName; }; }))
+    (_: value: callPackage package
+      (value // {
+        inherit src version openasar withOpenASAR;
+        unzip = unzipFix;
```

Comment:
```suggestion
```

Reply:
Ok, I've moved `unzipFix` to the insides of `openasar.nix`. Is it ok to leave it in the derivation scope, or is it better to constrain it to `postPatch`'s scope?